### PR TITLE
Audit and Authentication Policy Change Events

### PR DESCRIPTION
**Pull request by janniten**
## What does this PR do?

* Adds support for _**Audit and Authentication Policy Change Events**_

**Audit Audit   Policy Change** | **Audit Authentication Policy Change**
-- | --
https://docs.microsoft.com/en-us/windows/security/threat-protection/auditing/audit-audit-policy-change | https://docs.microsoft.com/en-us/windows/security/threat-protection/auditing/audit-authentication-policy-change
4719(S): System audit policy was changed. | 4670(S): Permissions on an object were changed
4817(S): Auditing settings on object were   changed. | 4706(S): A new trust was created to a domain.
4902(S): The Per-user audit policy table   was created. | 4707(S): A trust to a domain was removed.
4906(S): The CrashOnAuditFail value has   changed. | 4716(S): Trusted domain information was modified.
4907(S): Auditing settings on object were   changed. | 4713(S): Kerberos policy was changed.
4908(S): Special Groups Logon table   modified. | 4717(S): System security access was granted to an account.
4912(S): Per User Audit Policy was   changed. | 4718(S): System security access was removed from an account.
4904(S): An attempt was made to register   a security event source. | 4739(S): Domain Policy was changed.
4905(S): An attempt was made to   unregister a security event source. |  

Note:  Although processing of  Event 4715 (The audit policy (SACL) on an object was changed) seems to be identical to 4670, event 4715 was not included due I was not able to generate an example event.

For events where exists information of DACLs or SACLs those ACL are translated from the SDDL (https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/f4296d69-1c0f-491f-9587-a960b292d070) to a human-readable from. For example:

image

* Adds **related.ip** information in existing events where source.ip is present.

## Why is it important?
-  Auditing the changes in policies and event sources is crucial when we want to have a strong security monitoring system. Monitor these kinds of events are also important when we address compliance (SOX, PCI. HIPAA, etc )

- The related.ip information is useful when we want to pivot data between different sources. For example
      Fortinet Event (37141) indicating a user is connected to a VPN SSL when tunnelip is the asigned address.  Tunnelip is also in the related.ip field
      Windows Event 4624 indicating a windows login from a source.ip. If we have source.ip in the related.ip it is easy to match the user connected through VPN with a windows logon

## Checklist

- [x ] My code follows the style guidelines of this project
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [x ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by andrewkroh** [comment]
jenkins, test this

**Comment by janniten** [comment]
I confirm 👍

**Comment by janniten** [comment]
I still believe that those kind of events are important.
@andrewkroh what do you think?

Thank you

**Comment by andrewkroh** [comment]
💯 agree. Sorry it's taking so long to look these changes over.

**Comment by janniten** [comment]
New ECS 1.7.0 event category configuration used for auditory configuration and changes

https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html#ecs-event-category-configuration

**Comment by janniten** [comment]
Events related to audit configuration/changes are relevant when talking about defense evasion and also when you are under  strong security regulations.
New ECS 1.7.0 event category configuration used for auditory configuration and changes of this PR and in events from older PRs

**Comment by janniten** [comment]
Pinging @elastic/security-external-integrations , @andrewkroh 
Do you consider that this PR is still relevant? If not I can close it
Thank you!

**Comment by leehinman** [comment]
@janniten do you mind if I rebase off master and fix some conflicts.  There are a bunch in eventActionTypes, to support multiple values for event.category and event.type those are now arrays, so conflicts like:

```
"4697": [["iam", "configuration"], ["admin", "change"],"service-installed"],
vs
"4697": ["iam","admin","service-installed"]
```

**Comment by janniten** [comment]
> @janniten do you mind if I rebase off master and fix some conflicts. There are a bunch in eventActionTypes, to support multiple values for event.category and event.type those are now arrays, so conflicts like:
> 
> ```
> "4697": [["iam", "configuration"], ["admin", "change"],"service-installed"],
> vs
> "4697": ["iam","admin","service-installed"]
> ```

Hi @leehinman. No problem! Go ahead!
Thank you

**Comment by janniten** [comment]
> > @janniten do you mind if I rebase off master and fix some conflicts. There are a bunch in eventActionTypes, to support multiple values for event.category and event.type those are now arrays, so conflicts like:
> > ```
> > "4697": [["iam", "configuration"], ["admin", "change"],"service-installed"],
> > vs
> > "4697": ["iam","admin","service-installed"]
> > ```
> 
> Hi @leehinman. No problem! Go ahead!
> Thank you

I use in array values un order to suppport múltiples categoriew/types, but It is added later in code the when process the event.
It is better to define It in the way you propose.
Thank you!

**Comment by leehinman** [comment]
@janniten I couldn't push to your repo so I uploaded my changes here https://github.com/leehinman/beats/tree/janniten_aa_policy_change

Can you take a quick look and see if that looks right?  The intent was just address the eventActionTypes and to add golden files.

**Comment by janniten** [comment]
> @janniten I couldn't push to your repo so I uploaded my changes here https://github.com/leehinman/beats/tree/janniten_aa_policy_change
> 
> Can you take a quick look and see if that looks right? The intent was just address the eventActionTypes and to add golden files.

@leehinman , LGTM
I've noticed that there are some events defined in the eventActionTypes but not being processed (events 4616,4657 for example). Once finishished this PR I can add those ones.
Thank you!

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 1]
```diff
@@ -1495,12 +1739,118 @@ var security = (function () {
...
+        }
+    };
+
     var addSessionData = new processor.Chain()
         .Convert({
             fields: [
                 {from: "winlog.event_data.AccountName", to: "user.name"},
                 {from: "winlog.event_data.AccountDomain", to: "user.domain"},
                 {from: "winlog.event_data.ClientAddress", to: "source.ip", type: "ip"},
+                {from: "winlog.event_data.ClientAddress", to: "related.ip"},
```
Comment (leehinman) [inline x-pack/winlogbeat/module/security/config/winlogbeat-security.js:1853]: I think we want to add `type: "ip"`, I know we had at least one bug with this for with `source.ip`

File: x-pack/winlogbeat/module/security/config/winlogbeat-security.js [hunk 2]
```diff
@@ -1635,6 +2005,7 @@ var security = (function () {
                 {from: "winlog.event_data.ProcessId", to: "process.pid", type: "long"},
                 {from: "winlog.event_data.ProcessName", to: "process.executable"},
                 {from: "winlog.event_data.IpAddress", to: "source.ip", type: "ip"},
+                {from: "winlog.event_data.ClientAddress", to: "related.ip"},
```
Comment (leehinman) [inline x-pack/winlogbeat/module/security/config/winlogbeat-security.js:2008]: I think we want to add `type: "ip"`

**Review by leehinman — CHANGES_REQUESTED** [review]

**Review by leehinman — APPROVED** [review]

**Comment by leehinman** [comment]
run tests

**Comment by andrewkroh** [comment]
The new event IDs from this PR will need added to the list in https://www.elastic.co/guide/en/beats/winlogbeat/master/winlogbeat-module-security.html#winlogbeat-module-security.

**Comment by janniten** [comment]
> The new event IDs from this PR will need added to the list in https://www.elastic.co/guide/en/beats/winlogbeat/master/winlogbeat-module-security.html#winlogbeat-module-security.

Hi @andrewkroh. Done it https://github.com/elastic/beats/pull/23674

**Comment by janniten** [comment]
> The new event IDs from this PR will need added to the list in https://www.elastic.co/guide/en/beats/winlogbeat/master/winlogbeat-module-security.html#winlogbeat-module-security.

@andrewkroh , @leehinman Doc updated in https://github.com/elastic/beats/pull/23674

**Pull request merged**